Mixpanel/VariantAction.swift

```diff
@@ -0,0 +1,395 @@
+                guard let type = argumentTuple[1] as? String else {
+                    continue
+                }
+                if let transformedArg = VariantAction.transformValue(argumentTuple[0], to: type) {
+                    if transformedArg is NSValue {
+                        doesHaveNSValue = true
+                    } else if transformedArg is NSNull {
+                        doesHaveNSNullValue = true
+                    }
+                    transformedArgs.append(transformedArg)
+                }
+            }
+        }
+
+        for object in objects {
+            var retValue: AnyObject? = nil
+            let method: Method!
+            if object is AnyClass {
+                method = class_getClassMethod(object as! AnyClass, selector)
+            } else {
+                method = class_getInstanceMethod(type(of: object), selector)
+            }
+
+            guard method != nil else {
+                Logger.error(message: "Could not find a method with that selector value")
+                continue
+            }
+            let numArguments = method_getNumberOfArguments(method)
+            guard (Int(numArguments) - 2) >= transformedArgs.count else {
+                Logger.error(message: "Wrong number of arguments to invoke for selector")
+                continue
+            }
+            if doesHaveNSValue {
+                let implementation = method_getImplementation(method)
+                retValue = extractAndRunMethodFromSelector(selector: selector,
+                                                           implementation: implementation,
+                                                           object: object,
+                                                           args: transformedArgs)
+            } else if !doesHaveNSNullValue {
+                var unmanagedObject: Unmanaged<AnyObject>! = nil
+                if transformedArgs.isEmpty {
+                    unmanagedObject = object.perform(selector)
+                } else if transformedArgs.count == 1 {
+                    unmanagedObject = object.perform(selector, with: transformedArgs[0])
+                } else if transformedArgs.count == 2 {
+                    unmanagedObject = object.perform(selector, with: transformedArgs[0], with: transformedArgs[1])
+                } else {
+                    Logger.warn(message: "We do not support selectors that get more than 2 values")
+                }
+
+                if VariantAction.gettersForSetters.values.contains(selector) {
+                    retValue = unmanagedObject?.takeUnretainedValue()
+                }
+            }
+            targetRetValuePairs.append((object, retValue))
+        }
+        return targetRetValuePairs
+    }
+
+    class func extractAndRunMethodFromSelector(selector: Selector, implementation: IMP?, object: AnyObject, args: [Any]) -> AnyObject? {
+        if selector.description == "setImage:forState:" {
```

Comment:
is there no better way to do this in swift? seems unfortunate that supporting extra changes requires sdk updates